src/Campaigns/Blocks/CampaignComments/Controller/BlockRenderController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Give\Campaigns\Blocks\CampaignComments\Controller;
+
+use Give\Campaigns\Blocks\CampaignComments\DataTransferObjects\BlockAttributes;
+use Give\Campaigns\CampaignDonationQuery;
+use Give\Campaigns\Repositories\CampaignRepository;
+use Give\Donations\ValueObjects\DonationMetaKeys;
+use Give\Framework\Support\Facades\Scripts\ScriptAsset;
+
+/**
+ * @unreleased
+ */
+class BlockRenderController
+{
+    /**
+     * @unreleased
+     */
+    public function render(array $attributes): string
+    {
+        $blockAttributes = BlockAttributes::fromArray($attributes);
+
+        $encodedAttributes = json_encode($blockAttributes->toArray());
+
+        $this->loadScripts($blockAttributes);
```

Comment:
Instead of loading assets on the backend, this is a new approach we are using now ([block.json](https://github.com/impress-org/givewp/blob/b1b67c0485dc351f5e00cfe84894ceb2a2eab9b1/src/Campaigns/Blocks/CampaignGoal/block.json#L21)). With this approach you don't need `loadScripts` method.

Reply:
Hey yea thanks, I've been having difficulties localizing the window without registering it here. Any suggestions or thoughts?

Reply:
Hey Joshua, I'd suggest registering a new route - `/campaign/(?P<id>\d+)/comments` and then fetch data from the block app. For data fetching you can use `useSWR` which is great. Here is an example of how to use it https://github.com/impress-org/givewp/blob/eab7e73280c43290913249938cd45abb7afb3566/src/Campaigns/Blocks/DonateButton/edit.tsx#L32

Also, the current approach might not be the best if we want to introduce pagination for comments. Imagine there are hundreds of comments, which is unlikely but, the current approach won't be that efficient because all comments data will be loaded every single time. And if there are multiple blocks on the same page, but from different campaigns, every single one of them will use the same global object `GiveCampaignCommentsBlockWindowData`.

Reply:
@alaca I think that's a good suggestion for now, however in the future we should consider using the donations endpoint to get the comments (when those endpoints are ready).  We can then filter by `campaignId` and [HAL](https://stateless.group/hal_specification.html) to include donor data, checkout the slack thread on this topic https://lw.slack.com/archives/C04SLRDD9CK/p1738879925746179